test/extended/threads/atomic_with_offsets.wast

```diff
@@ -152,3 +152,55 @@
 (invoke "initOffset" (i64.const 0xffffffffffffffff) (i32.const 24000))
 (assert_return (invoke "i64.atomic.rmw32.cmpxchg_u" (i32.const 21169) (i64.const 0xffffffff) (i64.const 0xbeefbeefbeefbeef)) (i64.const 0xffffffff))
 (assert_return (invoke "i64.atomic.load" (i32.const 23980)) (i64.const 0xffffffffbeefbeef))
+
+(invoke "initOffset" (i64.const 0x1111111111111111) (i32.const 2872))
+(assert_return (invoke "i32.atomic.rmw16.cmpxchg_u" (i32.const 47) (i32.const 0x01ff1111) (i32.const 0xcafecafe)) (i32.const 0x1111))
+(assert_return (invoke "i64.atomic.load" (i32.const 2852)) (i64.const 0x1111111111111111))
+
+(invoke "initOffset" (i64.const 0x1111111111111111) (i32.const 8216))
+(assert_return (invoke "i64.atomic.rmw8.cmpxchg_u" (i32.const 5389) (i64.const 0x1000000000000000) (i64.const 0x4242424242424242)) (i64.const 0x11))
```

Comment:
The arguments are address, original value, new value? If the replace would be successful, the 0x42 value would be written?

Reply:
Yes exactly, address, expected original value, and new value.
If the value in memory and the expected original value match (which is not the case for this particular test case), the new value (0x42 in this case) is written to memory, and the original value is returned (0x11 in this case)